### DISCUSSION TITLE
How us use a command inside of nest scheduler?

### DISCUSSION THREAD

**Discussion by 9x3l6** [jmcdo29/nest-commander#848] [Q&A] [answered]
How to inject command into a service or processor or scheduler to run command from there?

**Comment by jmcdo29** [comment]
Why would you need to run a CLI command from the processor/scheduler instead of using the service directly?

**Comment by 9x3l6** [comment]
So to not write duplicate code, I'd like to run the command from a scheduler in a cron while my nest application is running, and I want to be able to run the commands manually when I need

**Reply by jmcdo29** [reply, marked as answer]
Well, to run a command from within a node application you use the `child_process` package from Node and `spawn` or `exec` a new shell to run the command.

**Comment by 9x3l6** [comment]
Okay, that works. I just thought maybe there would be a way to `@InjectCommand(BasicCommand)` or something like that. It would probably consume more memory to run than a `child_process`, am I right?

**Reply by jmcdo29** [reply]
In the end, commands are just `@Injectable()` classes (well `@Command()`, but that just sets metadata that nest-commander makes use of to parse options) so you could techincally inject the command and call `.run` yourself, but you won't get any of the `@Option()` methods to parse the options for you.

**Reply by 9x3l6** [reply]
Can you show me code example say with `BasicCommand` example that you have? Cuz I tried before asking and it get errors

**Reply by jmcdo29** [reply]
So long as the `BasicCommand` is in the `providers` array, or it is in the `exports` of a module that is in the `imports` there shouldn't be any problems.

```ts
@Module({
  imports: [BasicModule], // has `BasicCommand` in the `exports` and `providers` arrays
  providers: [SomeProcessor],
})
export class SomeModule {}

@Processor('some event')
export class SomeProcessor {
  constructor(private readonly basicCommand: BasicCommand) {}

  @Process()
  async processEvent() {
    await this.basicCommand.run(inputsStringArray, optionsObject);
  }
}
```
In theory, this would do fine

**Reply by 9x3l6** [reply]
That works great, thanks for your help
